gobdd_test.go

```diff
@@ -119,30 +147,43 @@ func fail(ctx context.Context) (context.Context, error) {
 	return ctx, errors.New("the step should never be executed")
 }
 
+func failure(ctx context.Context) (context.Context, error) {
+	return ctx, errors.New("the step failed")
+}
+
+func panics(_ context.Context) (context.Context, error) {
+	panic(errors.New("the step panicked"))
+}
+
 func pass(ctx context.Context) (context.Context, error) {
 	return ctx, nil
 }
 
 type mockTester struct {
 	fatalCalled int
+	errors      []string
 }
 
-func (m mockTester) Log(...interface{}) {
+func (m *mockTester) Log(...interface{}) {
```

Comment:
why did you put pointers here? If the function doesn't change anything I never put pointers :) https://golang.org/doc/effective_go.html#pointers_vs_values

Reply:
Because, in most cases, anyone adding an implementation would be changing something. Value receivers would quietly prevent the changes from persisting, leading to subtle bugs. I got bit by this while working on this bug, so I made sure the next guy wouldn't be. 

Also, [Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments#receiver-type) says

>  If in doubt, use a pointer